src/gs-category-page.c

```diff
@@ -230,6 +235,97 @@ copy_os_to_usb_button_cb (GtkButton *button, GsCategoryPage *self)
 			 self->cancellable);
 }
 
+static gchar*
+removable_destination_dup (GsCategoryPage *self)
+{
+	/* first one is the most-recently added mount on a removable disk */
+	if (self->copy_dests)
+		return g_strdup (self->copy_dests->data);
+
+	return NULL;
+}
+
+static void
+gs_category_page_os_get_copyable_cb (GObject *source_object,
+				     GAsyncResult *res,
+				     gpointer user_data)
+{
+	GsPluginLoader *plugin_loader = GS_PLUGIN_LOADER (source_object);
+	GsCategoryPage *self = GS_CATEGORY_PAGE (user_data);
+	g_autoptr(GError) error = NULL;
+	gboolean copyable;
+
+	copyable = gs_plugin_loader_job_os_get_copyable_finish (plugin_loader,
+								res, &error);
+	gtk_widget_set_visible (self->usb_action_box, copyable);
+}
+
+static void
+gs_category_page_update_os_copy_button (GsCategoryPage *self)
+{
+	g_autofree gchar* copy_dest = NULL;
+	g_autoptr(GsPluginJob) plugin_job = NULL;
+
+	if (self->category == NULL ||
+	    g_strcmp0 (gs_category_get_id (self->category), "usb") != 0) {
+		gtk_widget_set_visible (self->usb_action_box, FALSE);
+		return;
+	}
+
+	gtk_widget_set_visible (self->usb_action_box, TRUE);
+	gtk_widget_set_sensitive (self->copy_os_to_usb_button, FALSE);
+	gtk_widget_set_visible (self->copy_os_to_usb_button, FALSE);
+
+	if (self->copying)
+		return;
+
+	copy_dest = removable_destination_dup (self);
```

Comment:
Why does `removable_destination_dup()` need to `g_strdup()` the string? The duplicate is only used locally here, then freed.